### DISCUSSION TITLE
'Bare Metal' installation on a Raspberry

### DISCUSSION THREAD

**Discussion by Armag3ddon** [jonaswinkler/paperless-ng#978] [Support] [answered]
Hi,

I am trying to go for the 'Bare Metal Route' as described by the setup guide. I am intending to run paperless-ng on a Raspberry Pi 4. The FAQ tells me:
_If you decide to got with the bare metal route, be aware that some of the python requirements do not have precompiled packages for ARM / ARM64. Installation of these will require additional development libraries and compilation will take a long time._

I've run the requirements.txt as described: sudo -Hu paperless pip3 install -r requirements.txt and it ended in an error message (as expected, I guess).
Now, the FAQ hints to me that there could be a list of requirements that I need to compile myself. Do you indeed have the knowledge of which libraries are missing now or do I have to manually go through the list of requirements and check which ones were installed and which ones weren't?

Thank you in advance!

**Comment by Armag3ddon** [comment, marked as answer]
Well, I figured it out on my own.

For people in the future who stumble upon this problem:
The culprit seems to be ocrmypdf or rather pikepdf which doesn't have a package ready for the Pi.

In order to install ocrmypdf do the following steps (I did all that in /opt/paperless):
Some of the packages I install here will be already be installed by following the docs. I just copied this from where I found it originally.

```
sudo apt update
sudo apt dist-upgrade 
sudo apt install ghostscript libxml2 tesseract-ocr tesseract-ocr-deu pngquant unpaper
sudo apt install leptonica-progs libleptonica-dev automake libtool zlib1g-dev libjpeg-dev
sudo apt install python3 python3-pip libxml2-dev libxslt1-dev libffi-dev
git clone https://github.com/agl/jbig2enc
cd jbig2enc
./autogen.sh
./configure && make
sudo make install
cd .. 
git clone https://github.com/qpdf/qpdf 
cd qpdf
./configure && make
sudo make install
sudo ldconfig
cd ..
sudo pip3 install --upgrade pip
sudo -Hu paperless pip3  install pybind11
sudo -Hu paperless pip3 install ocrmypdf
```

Do that before
_sudo -Hu paperless pip3 install -r requirements.txt_
or, if you already did run that and stumbled into the problem like me, simply run the line again and all other dependencies should install just fine.
It'll require about 500MB of disc space and take like 15 minutes.

Paperless-ng is up and running. 😁

**Comment by Samganjii** [comment]
sudo apt update
sudo apt upgrade
sudo apt install python3 python3-pip python3-dev imagemagick fonts-liberation optipng gnupg libpq-dev libmagic-dev mime-support
sudo apt install ghostscript unpaper icc-profiles-free qpdf liblept5 libxml2 pngquant zlib1g tesseract-ocr tesseract-ocr-deu
sudo apt install leptonica-progs libleptonica-dev automake libtool zlib1g-dev libjpeg-dev
sudo apt install python3 python3-pip libxml2-dev libxslt1-dev libffi-dev
sudo apt install libatlas-base-dev libxslt1-dev
sudo apt install redis
systemctl start redis-server
systemctl enable redis-server
 
sudo apt install postgresql
sudo -u postgres psql
postgres=# create database mydb;
postgres=# create user myuser with encrypted password 'mypass';
postgres=# grant all privileges on database mydb to myuser;
\q
sudo adduser "paperless" --system --home /opt/paperless --group
cd /opt/paperless/
sudo -u paperless wget https://github.com/jonaswinkler/paperless-ng/releases/download/ng-1.4.5/paperless-ng-1.4.5.tar.xz
sudo -u paperless tar -xvf paperless-ng-1.4.5.tar.xz
sudo -u paperless mv /opt/paperless/paperless-ng/* /opt/paperless/
sudo rm paperless-ng-1.4.5.tar.xz
sudo nano ./paperless.conf
sudo -u paperless mkdir ./consume
sudo -u paperless mkdir ./data
sudo -u paperless mkdir ./media
sudo apt install git
sudo su
git clone https://github.com/agl/jbig2enc
cd jbig2enc
./autogen.sh
./configure && make
sudo make install
cd ..
git clone https://github.com/qpdf/qpdf
cd qpdf
./configure && make
exit
sudo make install
sudo ldconfig
cd ..
export PATH=$PATH:/opt/paperless/.local/bin/
sudo pip3 install --upgrade pip
sudo -Hu paperless pip3  install pybind11
sudo -Hu paperless pip3 install ocrmypdf
sudo -Hu paperless pip3 install --upgrade pip
sudo -Hu paperless pip3 install -r requirements.txt
cd /opt/paperless/src/
sudo -Hu paperless python3 manage.py migrate
sudo -Hu paperless python3 manage.py createsuperuser
sudo -Hu paperless python3 manage.py runserver
[dann die drei systemd scripte aus /opt/paperless/scripts/ in /usr/lib/systemd/system kopieren]
sudo systemctl start paperless-webserver.service
sudo systemctl enable paperless-webserver.service
sudo systemctl enable paperless-scheduler.service
sudo systemctl enable paperless-consumer.service
sudo systemctl start paperless-scheduler.service
sudo systemctl start paperless-consumer.service
